### ISSUE TITLE
Inheritance of classes with a qualifier parameter

### ISSUE THREAD

**Issue by smillst** [typetools/checker-framework#3400]
If a class extends a `@HasQualifierParameter` class (or implements a
`@HasQualifierParameter` interface), then that class must also be marked
`@HasQualifierParameter`. A particular subclass should be able to specify which qualifier
parameter should be used for the super class.  (This is similar to `class StringList implements List<String> {...}`.) For example,
```java
 @HasQualifierParameter(Tainted.class)
 class Buffer {
     void append(@PolyTainted Buffer this, @PolyTainted String s) { ... }
 }
 @HasQualifierParameter(Tainted.class)
 @Tainted class TaintedBuffer extends @Tainted Buffer {
     @Override
     void append(@Tainted TaintedBuffer this, @Tainted String s) { ... } // legal override
 }
```
`@Untainted TaintedBuffer` is an invalid type.

**Comment by t-rasmud** [comment]
#3832 makes `@HasQualifierParameter` an inherited type annotation addressing the requirement: "If a class extends a `@HasQualifierParameter` class (or implements a `@HasQualifierParameter` interface), then that class must also be marked
`@HasQualifierParameter`. "
The requirement "A particular subclass should be able to specify which qualifier parameter should be used for the super class." remains to be addressed.
